src/api/java/com/ldtteam/perviaminvenire/api/pathfinding/AbstractPathJob.java

```diff
@@ -395,7 +401,9 @@ protected Path search()
 
             calculationData.onNodeConsumed(currentNode.pos);
 
-            if (isAtDestination(currentNode))
+            // if xz restricted then disallow pathing to end outside given restriction
+            if ((!xzRestricted || (currentNode.pos.getX() >= minX && currentNode.pos.getX() <= maxX && currentNode.pos.getZ() >= minZ
```

Comment:
I thought about this change a bit more.
And on one side I like the idea here. But I fear that backwards compatibility maybe broken.

We basically have two scenarios.

1. We want the target to be inside a given space.
2. We want the pathing to only happen inside a given space.

Now both are valid options, and previously only 2. was implemented. When this PR is merged 2. is not possible anymore and 1. becomes the logic that is followed. Since both options have a reason to exist here, I think it would be a good idea if you added a toggle for this. Don't you think?

Reply:
Yeah, was my idea too, gonna add boolean switch if pathing outside restriction should be possible or not

Reply:
talking about restrictions in dms with Ray, I would say it's really hard to create a fixed border for 1. - we rather should use a better heuristic that will try to stay inside RA, eg. for lumberjack I've chosen as a goal a random pos in the furthest plane and used xz square metric for heuristic, normally you would probably choose a middle of RA as goal and also do that xz square metric

Reply:
we can also create a new search-break if eg. last 100 nodes were not in RA

Reply:
I'd just give the pathing itself the option to limit the explored nodes to a given area, and let the PathJob deal with any case by case logic where we could do stuff like give the pathing a larger area, but only allow destinations within a certain smaller area etc. Would use that aswell for backing/increasing search ranges and such

Reply:
For your treesearch you'd then give the pathing the tree area + leeway to walk around where it can moves in, and the job would only accept destinations within the tree area

Reply:
I've been talking with @Nightenom about this. I think the best way is actually have the search only happen within the area, starting at a point within that area and we just have the lumberjack first path to a random point within the area and then search from there.